spec/statsd/serialization/stat_serializer_spec.rb

```diff
@@ -108,6 +108,20 @@
         expect(input).to eq 'somecount:|@test'
       end
     end
+
+    context "when metric name is a symbol" do
+      it 'formats correctly without error' do
+        input = :'somecount::test'
+        output = subject.format(input, 1, 'c')
+        expect(output).to eq 'somecount.test:1|c'
+        expect(input).to eq :'somecount::test'
+
+        input = :'somecount:|@test'
+        output = subject.format(input, 1, 'c')
+        expect(output).to eq 'somecount___test:1|c'
+        expect(input).to eq :'somecount:|@test'
```

Comment:
Minor: It's a bit weird to assert on the input being unchanged since symbols are immutable? :eyes: 

```suggestion

        input = :'somecount:|@test'
        output = subject.format(input, 1, 'c')
        expect(output).to eq 'somecount___test:1|c'
```